packages/evm-adapter/src/approveTokenAmount.ts

```diff
@@ -0,0 +1,18 @@
+import { JsonRpcSigner, TransactionResponse } from '@ethersproject/providers';
+import { Contract } from 'ethers';
+import { erc20 } from './abi';
+import { contractAddress } from './constants';
+
+export const approveTokenAmount = async (
```

Comment:
from the title of this I would assume the function returns boolean but it returns TransactionResponse, maybe call it more descriptive

Reply:
It does exactly what is named, it doesn't prepare transaction but runs approval on chain, but yes it can return boolean

Reply:
OK